extensions/amp-story/0.1/audio.js

```diff
@@ -0,0 +1,429 @@
+    // Calculate the tree depth of the specified element.
+    let depth = 0;
+    for (let el = element; el; el = el.parentElement) {
+      depth++;
+    }
+
+    // The depth of background audio elements is one less, since we add the
+    // <audio> tag as a child of the element on which the background-audio
+    // attribute is specified.
+    if (element.classList.contains(BACKGROUND_AUDIO_ELEMENT_CLASS_NAME)) {
+      depth--;
+    }
+
+    return depth;
+  }
+
+  /**
+   * @return {!Element} The element causing audio to be played.
+   */
+  getSourceElement() {
+    return this.sourceElement_;
+  }
+
+  /**
+   * @return {!Element}
+   */
+  getDepth() {
+    return this.depth_;
+  }
+
+  /**
+   * Loads the resources necessary to play this item.  Should be a no-op if
+   * called when the item is already loaded.
+   * @return {!Promise} A promise that is resolved once the resource has been
+   *     loaded.
+   */
+  load() {
+    return Promise.resolve();
+  }
+
+  /**
+   * @return {boolean} true, if this item's resources have been loaded.
+   */
+  isLoaded() {
+    return false;
+  }
+
+  /**
+   * Causes this item to start playing if it was not already playing.  load()
+   * will be called before play() in all cases.  play() should be a no-op if the
+   * item is already being played.
+   */
+  play() {}
+
+  /**
+   * Sets the volume of this item to the specified volume, over the specified
+   * duration of time.
+   *
+   * @param {number} unusedVolume A volume from 0.0 (silent) to 1.0 (loudest).
+   * @param {number} unusedDurationMs The duration over which the new volume
+   *     should be achieved.
+   * @param {!function(number): number} unusedEasingFn The easing function which
+   *     describes the curve the volume should be modified.
+   */
+  setVolume(unusedVolume, unusedDurationMs, unusedEasingFn) {}
+
+  /**
+   * Causes this item to stop playing if it was playing.  stop() should be a
+   * no-op if the item is already stopped.
+   */
+  stop() {}
+
+  /**
+   * Unloads the resources associated with this item.  Can be called to free up
+   * resources.  Should be a no-op if called when the item is not yet loaded.
+   */
+  unload() {}
+
+  /**
+   * Mutes this item, without affecting its volume or play state.
+   */
+  mute() {}
+
+  /**
+   * Unmutes this item, without affecting its volume or play state.
+   */
+  unmute() {}
+
+  /**
+   * @return {boolean} true, if this item is muted; false otherwise.
+   */
+  isMuted() {}
+}
+
+
+/**
+ * An HTMLMediaElement that potentially has audio.
+ */
+class MediaElementPlayable extends Playable {
+  constructor(win, element) {
+    super(win, element);
+    this.element_ = element;
+  }
+
+  /** @override */
+  isLoaded() {
+    return !!this.element_;
+  }
+
+  /** @override */
+  isPlaying() {
+    return !this.element_.paused;
+  }
+
+  /** @override */
+  play() {
+    this.element_.play();
+  }
+
+  /** @override */
+  setVolume(volume, durationMs, unusedEasingFn) {
```

Comment:
Let's document somewhere the iOS's restriction on setting volume and how that's handled.